opc/resource_lbaas_certificate.go

```diff
@@ -0,0 +1,154 @@
+			},
+			"certificate_body": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIsPEMFormat,
+			},
+			"certificate_chain": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIsPEMFormat,
+			},
+			"private_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIsPEMFormat,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"SERVER",
+					// "TRUSTED", TODO disable TRUSTED certs due to deletion issue
+				}, true),
+			},
+
+			// Read only attributes
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSSLCertificateCreate(d *schema.ResourceData, meta interface{}) error {
+
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	sslCertClient := lbaasClient.SSLCertificateClient()
+
+	input := lbaas.CreateSSLCertificateInput{
+		Name:             d.Get("name").(string),
+		Certificate:      d.Get("certificate_body").(string),
+		CertificateChain: d.Get("certificate_chain").(string),
+		PrivateKey:       d.Get("private_key").(string),
+		Trusted:          d.Get("type").(string) == "TRUSTED",
+	}
+
+	info, err := sslCertClient.CreateSSLCertificate(&input)
+	if err != nil {
+		return fmt.Errorf("Error creating Load Balancer Server Pool: %s", err)
+	}
+
+	d.SetId(info.Name)
+	return resourceSSLCertificateRead(d, meta)
+}
+
+func resourceSSLCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	sslCertClient := lbaasClient.SSLCertificateClient()
+	name := d.Id()
+
+	result, err := sslCertClient.GetSSLCertificate(name)
+	if err != nil {
+		// SSLCertificate does not exist
+		if client.WasNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Server Pool %s: %s", d.Id(), err)
+	}
+
+	if result == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("certificate_body", result.Certificate)
+	d.Set("certificate_chain", result.CertificateChain)
+	d.Set("name", result.Name)
+	d.Set("state", result.State)
+	d.Set("uri", result.URI)
+
+	if result.Trusted {
+		d.Set("type", "TRUSTED")
+	} else {
+		d.Set("type", "SERVER")
+	}
+
+	return nil
+}
+
+func resourceSSLCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	sslCertClient := lbaasClient.SSLCertificateClient()
+	name := d.Id()
+
+	if _, err := sslCertClient.DeleteSSLCertificate(name); err != nil {
+		return fmt.Errorf("Error deleting SSLCertificate")
```

Comment:
Let's also write out the error when we fail to delete an SSLCertificate

Reply:
done